### ISSUE TITLE
Building error on Ubuntu 22.04 due to undefined symbol: PKCS12_SAFEBAG_free

### ISSUE THREAD

**Issue by alazarolop** [jeroen/openssl#103]
Hi, 

I am unable to build the openssl R package on WSL Ubuntu 22.04 using R downloaded from RStudio site:

```
$ uname -a
Linux DESKTOP-3UFB8R8 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

$ R --version
R version 4.2.0 (2022-04-22) -- "Vigorous Calisthenics"
Copyright (C) 2022 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)
````

I have installed and reinstalled openssl OS packages:

```
$ sudo apt reinstall libssl-dev libssl3
$ openssl version -a

OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)
built on: Thu May  5 08:04:52 2022 UTC
platform: debian-amd64
options:  bn(64,64)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -Wa,--noexecstack -g -O2 -ffile-prefix-map=/build/openssl-Ke3YUO/openssl-3.0.2=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -DOPENSSL_TLS_SECURITY_LEVEL=2 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2
OPENSSLDIR: "/usr/lib/ssl"
ENGINESDIR: "/usr/lib/x86_64-linux-gnu/engines-3"
MODULESDIR: "/usr/lib/x86_64-linux-gnu/ossl-modules"
Seeding source: os-specific
CPUINFO: OPENSSL_ia32cap=0xfeda32235f8bffff:0x9c27a9
```

When building the R package I get the following error:

```
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘openssl’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/R/4.2.0/lib/R/site-library/00LOCK-openssl/00new/openssl/libs/openssl.so':
  /opt/R/4.2.0/lib/R/site-library/00LOCK-openssl/00new/openssl/libs/openssl.so: undefined symbol: PKCS12_SAFEBAG_free
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/opt/R/4.2.0/lib/R/site-library/openssl’
```

How could I fix it?
Thanks

**Comment by jeroen** [comment]
Can you show the full installation log?

**Comment by alazarolop** [comment]
1. I installed following instructions in RStudio, so I need to build packages from source.

```
curl -O https://cdn.rstudio.com/r/ubuntu-${OS_VERSION}/pkgs/r-${R_VERSION}_1_amd64.deb
sudo gdebi r-${R_VERSION}_1_amd64.deb
```

2. Yes, I had `ccache` in Makevars. I erased it, but I got the same error. 
3. `/usr/local/lib` doesn't have any openssl folder inside. The same happens in `usr/local/include/openssl`.
4. I see what you meant on `/usr/local/lib` harcoded. But I wasn't able to locate it (as I said). I thought the problem could come from R build and installation process. Thus, I decided to create two new WSL (20.04 and 22.04) and follow the same installation process. I was able to install the openssl R package in 20.04, but it was throwing the issue in 22.04. I noticed 22.04 is using 3.0.2 version rather than the 1.1.1 version in 20.04. I wonder whether the problem could come from openssl building in 22.04 or maybe the R package not being able to cope with 3.0.2 (in Ubuntu 22.04)?

**Comment by jeroen** [comment]
What do you mean by "the same happens in usr/local/include/openssl"? Does that folder exist?

I use ubuntu 22.04 myself and many other systems with openssl 3.0.2 so the problem is definitely in your configuration.

**Comment by jeroen** [comment]
I just downloaded the same R as you did for ubuntu 22.04, but the configuration is very different. I think the problem happens in this line in your output:

```
ccache gcc-11 -shared -L/opt/R/4.2.0/lib/R/lib -L/usr/local/lib -o openssl.so aes.o base64.o bignum.o cert.o compatibility.o diffie.o envelope.o error.o hash.o info.o keygen.o keys.o onload.o openssh.o password.o pbkdf.o pem.o pkcs12.o pkcs7.o rand.o rsa.o signing.o ssl.o stream.o write.o x25519.o -fopenmp -lgomp -L/opt/R/4.2.0/lib/R/lib -lR
```

On my own ubuntu 22.04 machine I see this 

```
gcc -shared -L/opt/R/4.2.0/lib/R/lib -L/usr/local/lib -o openssl.so aes.o base64.o bignum.o cert.o compatibility.o diffie.o envelope.o error.o hash.o info.o keygen.o keys.o onload.o openssh.o password.o pbkdf.o pem.o pkcs12.o pkcs7.o rand.o rsa.o signing.o ssl.o stream.o write.o x25519.o -Lbcrypt -lstatbcrypt -l:libssl.so.3 -l:libcrypto.so.3 -L/opt/R/4.2.0/lib/R/lib -lR
```

It looks like your linker flags are completely gone. Are you overriding `PKG_LIBS` in your local config or something?

**Comment by jeroen** [comment]
@alazarolop has your problem been solved? Can you check if your settings are overriding `PKG_LIBS` or `LDFLAGS` or similar?

**Comment by alazarolop** [comment]
Hi and sorry for the late response, 

I tried to follow yours comments, but I couldn't figure out where the overriding was happening. 

Recently, I've tried again after upgrading all the packages (openssl included) and it worked out-of-the-box in a vanilla installation (no config changes in WSL Ubuntu 22.04), so I guess the SO package was involved in the error. I guess the ticket can be now closed. 
Thank you a lot for your support.
